powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/ActivationServiceBehavior.java

```diff
@@ -1066,13 +1060,27 @@ public PrepareActivationResponse prepareActivation(PrepareActivationRequest requ
                         protocolVersion);
             }
 
-            final String masterPrivateKeyBase64 = masterKeyPairEntity.getMasterKeyPrivateBase64();
-            final PrivateKey privateKey = keyConvertor.convertBytesToPrivateKey(Base64.getDecoder().decode(masterPrivateKeyBase64));
+            final PrivateKey privateKey;
+            if (temporaryKeyId == null) {
```

Comment:
The could would be much more readable with inverted logic `if (temporaryKeyId != null)`.